docs/guide_slash_and_interactions.rst

```diff
@@ -0,0 +1,297 @@
+
+Decorators
+----------
+Just like with text commands, we can use decorators to modify the behaviour of our slash commands.
+For example, we can use the :func:`app_commands.guild_only` decorator to make our slash command only work in guilds.
+
+.. code-block:: python
+
+    import discord
+
+    from redbot.core import commands, app_commands
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @app_commands.command()
+        @app_commands.guild_only()
+        async def hello(self, interaction: discord.Interaction):
+            await interaction.response.send_message("Hello World!", ephemeral=True)
+
+One of the more useful decorators is the :func:`app.commands.choices` decorator.
+This decorator allows us to specify a list of choices for a specific argument.
+This is useful for arguments that have a limited number of options.
+For example, we can use this to create a command that allows us to choose between two different colors.
+
+.. code-block:: python
+
+    from enum import Enum
+    from redbot.core import commands, app_commands
+
+    class Color(Enum):
+        Red = 1
+        Blue = 2
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @app_commands.command()
+        @app_commands.describe(color="The color you want to choose")
+        async def color(self, interaction: discord.Interaction, color: Color):
+            await interaction.response.send_message(f"Your color is {str(color.name)}", ephemeral=True)
+
+Check out the full reference of decorators on Discord.py's documentation `here <https://discordpy.readthedocs.io/en/stable/interactions/api.html#decorators>`__.
+
+
+Groups & Subcommands
+--------------------
+Slash commands can also be grouped together into groups and subcommands.
+These can be used to create a more complex command structure.
+
+.. note::
+    Unlike text command groups, top level slash command groups **cannot** be invoked.
+
+.. code-block:: python
+
+    import discord
+    
+    from redbot.core import commands, app_commands
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        group = app_commands.Group(name="zoo", description="Zoo related commands")
+
+        @group.command(name="add", description="Add an animal to the zoo")
+        @app_commands.describe(animal="The animal you want to add")
+        async def add_animal(self, interaction: discord.Interaction, animal: str):
+            await interaction.response.send_message(f"Added {animal} to the zoo", ephemeral=True)
+
+        @group.command(name="remove", description="Remove an animal from the zoo")
+        @app_commands.describe(animal="The animal you want to remove")
+        async def remove_animal(self, interaction: discord.Interaction, animal: str):
+            await interaction.response.send_message(f"Removed {animal} from the zoo", ephemeral=True)
```

Comment:
```suggestion
        zoo = app_commands.Group(name="zoo", description="Zoo related commands")

        @zoo.command(name="add", description="Add an animal to the zoo")
        @app_commands.describe(animal="The animal you want to add")
        async def zoo_add(self, interaction: discord.Interaction, animal: str):
            await interaction.response.send_message(f"Added {animal} to the zoo", ephemeral=True)

        @zoo.command(name="remove", description="Remove an animal from the zoo")
        @app_commands.describe(animal="The animal you want to remove")
        async def zoo_remove(self, interaction: discord.Interaction, animal: str):
            await interaction.response.send_message(f"Removed {animal} from the zoo", ephemeral=True)
```
Keep the variable names consistent with how text command groups are _usually_ done.